lib/internal/modules/esm/translators.js

```diff
@@ -228,11 +231,17 @@ function loadCJSModule(module, source, url, filename) {
       ],
       hostDefinedOptionId,           // hostDefinedOptionsId
       importModuleDynamically,       // importModuleDynamically
-    ).function;
+    );
   } catch (err) {
     enrichCJSError(err, source, filename);
     throw err;
   }
+  // Cache the source map for the cjs module if present.
+  if (compileResult.sourceMapURL) {
+    maybeCacheSourceMap(url, source, null, false, url, compileResult.sourceMapURL);
```

Comment:
The fifth argument `sourceURL` should be retrieved with V8 API `GetSourceURL()` or set as `undefined` (which would lead to a RegExp search on the content) so that `findSourceMap` can correctly recognize file names in stacktrace that have been translated by V8.

For instance, a script with the following content:

```js
console.trace('');
//# sourceURL=renamed-source-url
```

The output stack would be as follows regardless of what the file is actually named.
```
Trace
    at Object.<anonymous> (renamed-source-url:1:9)
```